### PR TITLE
chore: ab test for video in launch tutorial welcome screen

### PR DESCRIPTION
**Pull request by Harshita-mindfire**
This PR aims to add an AB test for 1 min feature highlights video in the Welcome screen displayed after running `Launch Tutorial` command. 

- used `webm` format for video, with `.mp4` there was no sound.

# Dendron Extended PR Checklist

### Basics

- [x] code should follow [Code Conventions](dev.process.code)
- [x] circular dependency check: make sure your code is not introducing new circular dependencies in plugin-core.  See [Avoiding Circular Dependencies](dev.process.code.best-practices).
- [x] sticking to existing conventions instead of creating new ones
  - eg: [if configuration for utilities are already in one module or package, add future utilities there as well](https://github.com/dendronhq/dendron/pull/1960#discussion_r786228021)

### Extended
- General
  - [x] check whether code be simplified
  - [x] check if similar function already exist in the codebase. if so, can it be re-used?
  - [x] check if this change adversely impact performance
- Operations
  - [x] when shipping this change, will it just work or will it introduce additional operational overhead due to complicated interface or known bugs?
- Architecture
  - [x] check if code is introducing changes on a foundational class or interface. if so, call for design review if needed
    - eg: [making changes to DNode](https://github.com/dendronhq/dendron/pull/2158#pullrequestreview-854689586)

### Basics
- [ ] if you are adding analytics related changes, make sure the [Telemetry](https://wiki.dendron.so/notes/84df871b-9442-42fd-b4c3-0024e35b5f3c.html) docs are updated

### Extended
- [~] can we track the performance of this change to know if it is _successful_?
  - eg: [see usage for export pod](https://github.com/dendronhq/dendron/pull/2190#pullrequestreview-855715612)

### Basics

- [ ] [Write Tests](dev.process.qa.test) 
- [ ] [Confirm existing tests pass](dev.process.qa.test)
- [x] [Confirm manual testing](dev.process.qa.test) 
- [ ] Common cases tested
- [ ] 1-2 Edge cases tested
- [ ] If your tests changes an existing snapshot, snapshots have been [updated](dev.process.qa.test)

## Docs
- [ ] if your change reflects documentation changes, also submit a PR to [dendron-site](https://github.com/dendronhq/dendron-site) and mention the doc PR link in your current PR
- [ ] does this change introduce a new or better way of doing things that others need to be aware of? if so, an async should be created and a process added in [Development](dev) or [Packages](pkg)

### Extended
- [~]  is this a developer BREAKING change? if another person cloning from this branch will need to adjust their dependencies or mental model of the architecture, then it is. if this is the case, make sure this is communicated according to [Close Loop](dev.process.close-loop)
  - eg: [breaking dev change due to new dependency](https://github.com/dendronhq/dendron/pull/2188#pullrequestreview-855696330)

### CODE REVIEW

**Comment by jonathanyeung** [comment]
As discussed offline, if we can't figure out a way to embed, I'd rather us not run this experiment rather than include the file as part of our extension binary.  I think that the binary size will adversely affect download rates and it's also not going to publish to ovsx as it stands (32 mb single file size limit).  @kevinslin - thoughts?

**Comment by Harshita-mindfire** [comment]
one possible workaround is uploading the video to google drive. This works fine in preview with a valid id
```html
<video preload="true" controls autoplay>
  <source
    src="https://drive.google.com/uc?export=download&amp;id=<insert-video-drive-id-here>"
    type="video/webm"
  />
</video>
```

File: packages/plugin-core/assets/dendron-ws/vault/welcome_video.html [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+  </div>
+    <br>
+    <div id="btn_container" class="container">
+      <a class="initialize_btn" id="btn_id" target="_blank" rel="nofollow"><span>Get Started</a>
+    </div>
+    
+    <div id="gif-ctn">
+      <video controls autoplay>
+        <source
+        src="https://org-dendron-public-assets.s3.amazonaws.com/images/dendron_feature_highlights.webm"
```
Comment (kevinslin) [inline packages/plugin-core/assets/dendron-ws/vault/welcome_video.html:108]: can we use the cloudfront link instead? will load faster > https://d2q204iup008xl.cloudfront.net/images/dendron_feature_highlights.webm

File: packages/plugin-core/src/WelcomeUtils.ts [hunk 2]
```diff
@@ -21,14 +23,33 @@ async function initWorkspace() {
...
-      "dendron-ws",
-      "vault",
-      "welcome.html"
+    let content: string;
+    const ABUserGroup = _2022_09_VIDEO_TUTORIAL_TEST.getUserGroup(
+      SegmentClient.instance().anonymousId
     );
-    const { content } = readMD(uri.fsPath);
+    let testgroup: string;
+    const version = Number(vscode.version.substring(0, 4)); // version: x.xx.x
```
Comment (kevinslin) [inline packages/plugin-core/src/WelcomeUtils.ts:31]: can we use the `semver` package to check for versions?

File: packages/plugin-core/assets/dendron-ws/vault/welcome_video.html [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+    <p class="subtitle">Note, organize, and reference any amount of information.</p>
+  </div>
+    <br>
+    <div id="btn_container" class="container">
+      <a class="initialize_btn" id="btn_id" target="_blank" rel="nofollow"><span>Get Started</a>
+    </div>
+    
+    <div id="gif-ctn">
+      <video controls autoplay>
+        <source
```
Comment (kevinslin) [inline packages/plugin-core/assets/dendron-ws/vault/welcome_video.html:107]: can we make this take the full width of the screen?

**Review by kevinslin — CHANGES_REQUESTED** [review]
video doesn't seem to load or play for me > https://www.loom.com/share/7b43a4b575e9431c82b686f3cb8a7e93

**Comment by Harshita-mindfire** [comment]
> video doesn't seem to load or play for me > 

It looks like it is still downloading. Do you have any errors in console/network?

**Review by kevinslin — APPROVED** [review]

**Pull request merged**